std/assembly/string.ts

```diff
@@ -661,12 +661,13 @@ export namespace String {
     export function encode(str: string, nullTerminated: bool = false): ArrayBuffer {
       var strOff = changetype<usize>(str);
       var strEnd = changetype<usize>(str) + <usize>changetype<BLOCK>(changetype<usize>(str) - BLOCK_OVERHEAD).rtSize;
-      var buf = __alloc(UTF8.byteLength(str, nullTerminated), idof<ArrayBuffer>());
+      var bufLen = <usize>UTF8.byteLength(str, nullTerminated);
+      var buf = __alloc(bufLen, idof<ArrayBuffer>());
+      var bufEnd = buf + bufLen - i32(nullTerminated);
```

Comment:
Makes me wonder if we have `usize(nullTerminated)` in portable?

Reply:
makes sense

Reply:
Seems we don't. Should probably add this as an alias of `u32` to be correct.